lib/Lib.MultiBuffer.fst

```diff
@@ -0,0 +1,92 @@
+module Lib.MultiBuffer
+
+open FStar.Mul
+module ST = FStar.HyperStack.ST
+open FStar.HyperStack
+open FStar.HyperStack.All
+
+open Lib.IntTypes
+open Lib.Sequence
+open Lib.Buffer
+open Lib.NTuple
+
+#set-options "--z3rlimit 30 --fuel 0 --ifuel 0"
+
+let live4 #a #len (h:mem) (b0 b1 b2 b3: lbuffer a len) =
+  live h b0 /\ live h b1 /\ live h b2 /\ live h b3
+
+let live8 #a #len (h:mem) (b0 b1 b2 b3 b4 b5 b6 b7: lbuffer a len) =
+  live h b0 /\ live h b1 /\ live h b2 /\ live h b3 /\ live h b4 /\ live h b5 /\ live h b6 /\ live h b7
+
+let internally_disjoint4 #len #a (b0 b1 b2 b3: lbuffer a len) =
+  disjoint b0 b1 /\ disjoint b0 b2 /\ disjoint b0 b3 /\
+  disjoint b1 b2 /\ disjoint b1 b3 /\ disjoint b2 b3
+
+let internally_disjoint8 #len #a (b0 b1 b2 b3 b4 b5 b6 b7: lbuffer a len) =
```

Comment:
FYI there's a library to do this automatically rather than write all pairwise combinations by hand: https://github.com/FStarLang/FStar/blob/master/ulib/FStar.BigOps.fsti -- see https://github.com/FStarLang/FStar/blob/master/ulib/LowStar.Monotonic.Buffer.fsti#L1004 for an example of usage